src/server/helper/kmsClientHelper.js

```diff
@@ -0,0 +1,91 @@
+import NyplDataApiClient from '@nypl/nypl-data-api-client';
+import aws from 'aws-sdk';
+
+const kmsClientHelper = (options) => {
+  const {
+    kmsEnvironment,
+    apiBase,
+    tokenUrl,
+    clientId,
+    clientSecret,
+  } = options;
+  const keys = [clientId, clientSecret];
+  const CACHE = {};
+  let decryptKMS;
+  let kms;
+
+  // If we want to use encrypted client id and secret strings, then we need to set up
+  // AWS and the KMS decryption function.
+  if (kmsEnvironment === 'encrypted') {
+    kms = new aws.KMS({
+      region: 'us-east-1',
+    });
+
+    decryptKMS = (key) => {
+      const params = {
+        CiphertextBlob: new Buffer(key, 'base64'),
+      };
+
+      return new Promise((resolve, reject) => {
+        kms.decrypt(params, (err, data) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(data.Plaintext.toString());
+          }
+        });
+      });
+    };
+  }
+
+  function client() {
+    // If we already instantiated the NYPL Api Data Client, then just return it.
+    if (CACHE.nyplApiClient) {
+      return Promise.resolve(CACHE.nyplApiClient);
+    }
+
+    // If we want to use encrypted client id and secret strings, then we first need to
+    // decrypt the strings and instantiate the NyplDataApiClient with those decrypted values.
+    if (kmsEnvironment === 'encrypted') {
+      return new Promise((resolve, reject) => {
+        Promise.all(keys.map(decryptKMS))
```

Comment:
`Promise.all` returns a promise, so instead of returning a `new Promise` on line 50, I think technically you could just `return Promise.all` (and then simply `return nyplApiClient` on line 64 & `throw` the error on 68)

Reply:
Thanks! I knew it looked odd and wanted to update it. Yes, it does return a promise so the outer one can be removed.